File: NationalSecurityCommunity/app/src/main/java/com/national/security/community/injection/component/AppComponent.java
package com.national.security.community.injection.component;


import com.national.security.community.App;
import com.national.security.community.helper.RealmHelper;
import com.national.security.community.helper.RetrofitHelper;
import com.national.security.community.injection.module.AppModule;
import com.national.security.community.injection.module.HttpModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, HttpModule.class})
public interface AppComponent {

    App getContext();  // 提供App的Context

    RetrofitHelper retrofitHelper();  //提供http的帮助类

    RealmHelper realmHelper();    //提供数据库帮助类
}

File: app/src/main/java/com/jmd_news_kotlin/di/component/AppComponent.kt
package com.jmd_news_kotlin.di.component

import com.jmd_news_kotlin.App
import com.jmd_news_kotlin.di.module.AppModule
import com.jmd_news_kotlin.di.module.HttpModule
import com.jmd_news_kotlin.net.ApiService
import com.jmd_news_kotlin.utils.RetrofitHelper
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, HttpModule::class))
interface AppComponent {

    fun getContext(): App
    fun retrofitHelper(): RetrofitHelper
    fun apiService(): ApiService
}